packages/opentelemetry-core/src/trace/instrumentation/basePluginUtils.ts

```diff
@@ -0,0 +1,48 @@
+/*!
```

Comment:
I guess I'm not totally convinced that a separate package for just utils would add much vs. having these in "core".

Since tracing in the Node/Web context is generally about requests of various kinds, and URL-based requests are pretty common given the HTTP protocol, I personally think these are fine utilities to have in `core`.

I wonder if they could be a bit more generically named, say to "stringMatches", "isIgnored" and "ignoredStrings", etc. Then non-URL based protocols like gRPC could still use it, and we could make the argument that ignore lists are a generic cross-plugin concept like timestamps that deserves to be in core given its relatively wide usefulness.

I guess I would see a `utils` package as just collecting lots of functions and especially if `core` ends up wanting to depend on `utils` then it's effectively part of core anyway.

I'm open though on this!

Reply:
I agree that calling it basePluginUtils is misleading. I might call it "stringMatchUtils" or similar.